runtime/src/iree-amd-aie/driver/xrt-lite/util.h

```diff
@@ -20,6 +20,14 @@ iree_status_t unimplemented_ok_status(Params...) {
 }
 
 template <typename... Params>
-void unimplemented_ok_void(Params...) {}
+void unimplemented_ok_void(Params...){}
+#ifndef NDEBUG
+#define IREE_HAL_XRT_LITE_CHECKED_VTABLE_CAST(base_value, vtable, subvalue_t) \
+  (IREE_HAL_ASSERT_TYPE(base_value, &vtable),                                 \
```

Comment:
FWIW `IREE_HAL_ASSERT_TYPE` should already be getting elided in a non-debug build. It would turn into 
```
while (false && (iree_hal_resource_is(....)) {
}
```
which the compiler "SHOULD" just be removed entirely.